lib/alexa.js

```diff
@@ -143,13 +144,13 @@ function ValidateRequest() {
         // Validate that this request originated from authorized source.
         if (handlerAppId && (requestAppId !== handlerAppId)) {
             console.log(`The applicationIds don\'t match: ${requestAppId} and ${handlerAppId}`);
-            return context.fail('Invalid ApplicationId: ' + handlerAppId);
+            return callback(new Error('Invalid ApplicationId: ' + handlerAppId));
```

Comment:
It's possible for callback to be undefined as it's an optional parameter see [callback parameter](http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html#nodejs-prog-model-handler-callback)

This could result in some skills breaking when making this update. It would also make this library incompatible for Node v0.10.x .

My suggestion is checking if callback was passed in and is a function then falling back to context.succeed() & fail()